src/cpp/utils.h

```diff
@@ -46,13 +48,40 @@ class ScopeGuard {
     std::function<void()> f_;
 };
 
-} // effil
+class Logger
```

Comment:
I think runtime logger is redrurant abstraction for this project. 
It will add extra expences - function calls, string formatting operations and file operations.